src/Microsoft.AspNetCore.All/build/PublishWithAspNetCoreTargetManifest.targets

```diff
@@ -0,0 +1,49 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <PublishWithAspNetCoreTargetManifest Condition="'$(PublishWithAspNetCoreTargetManifest)'==''">true</PublishWithAspNetCoreTargetManifest>
+  </PropertyGroup>
+
+<!--
+*****************************************************************************
+Target: PublishWithAspNetCoreTargetManifest
+Append the default ASP.NET Core runtime package store manifest during publish
+*****************************************************************************
+-->
+  <Target
+    Name="PublishWithAspNetCoreTargetManifest"
+    BeforeTargets="Publish"
+    DependsOnTargets="PrepareForPublish"
+    Condition="'$(PublishWithAspNetCoreTargetManifest)'=='true' and '$(PublishableProject)'=='true'" >
+
+    <PropertyGroup>
+      <ManifestRuntimeIdentifier>$(AspNetCoreTargetManifestRuntimeIdentifier)</ManifestRuntimeIdentifier>
+      <ManifestRuntimeIdentifier Condition="'$(AspNetCoreTargetManifestRuntimeIdentifier)'==''
+          and '$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Windows)))'
+          and '$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture)'=='X64'">win7-x64</ManifestRuntimeIdentifier>
+      <ManifestRuntimeIdentifier
+        Condition="'$(AspNetCoreTargetManifestRuntimeIdentifier)'==''
+          and '$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Windows)))'
+          and '$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture)'=='X86'">win7-x86</ManifestRuntimeIdentifier>
+      <ManifestRuntimeIdentifier
+        Condition="'$(AspNetCoreTargetManifestRuntimeIdentifier)'==''
+          and '$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::OSX)))'">osx.10.12-x64</ManifestRuntimeIdentifier>
+      <ManifestRuntimeIdentifier
+        Condition="'$(AspNetCoreTargetManifestRuntimeIdentifier)'==''
+          and '$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Linux)))'">linux-x64</ManifestRuntimeIdentifier>
+    </PropertyGroup>
+
+    <Error
+      Text="Could not resolve manifest runtime identifier. Please specify the the runtime identifier via the AspNetCoreTargetManifestRuntimeIdentifier property."
+      Condition="'$(ManifestRuntimeIdentifier)'==''" />
+
+    <Message
+      Text="Appending default ASP.NET Core runtime package store manifest for use during publish based for the runtime $(ManifestRuntimeIdentifier)."
+      Importance="normal"
+      Condition="'$(ManifestRuntimeIdentifier)'!=''" />
```

Comment:
Wasn't sure if it should be an error. If we couldn't resolve the manifest, maybe we should just give a warning and publish without manifest.